roles/jd-client/src/lib/mod.rs

```diff
@@ -170,7 +170,7 @@ impl JobDeclaratorClient {
                     },
                     _ = self.shutdown.notified().fuse() => {
                         info!("Shutting down gracefully...");
-                        std::process::exit(0);
+                        break 'outer;
```

Comment:
In `inner` loop, `break` is used to exit current(`inner`) loop. I would use `break` here as well to keep it consistent.

Reply:
aint this match arm in the `outer` loop?